Guide/Greedy/Stable Marriage/README.md

```diff
@@ -0,0 +1,294 @@
+
+Category: Greedy
+
+Difficulty: N/A
+
+## Problem
+
+### Overview
+
+Given _n_ men and _n_ women, where each person has ranked all members of the opposite sex in order of preference, marry the men and the women together such that there are no two people of opposite sex who would both rather have each other than their current partners. When there are no such pairs of people, the set of marriages is deemed stable.
+
+_From Wikipedia:_
+Stable marriage problem (also stable matching problem or SMP) is the problem of finding a stable matching between two equally sized sets of elements given an ordering of preferences for each element. A matching is a mapping from the elements of one set to the elements of the other set. A matching is not stable if:
+
+- There is an element A of the first matched set which prefers some given element B of the second matched set over the element to which A is already matched, and
+
+- B also prefers A over the element to which B is already matched.
+
+i.e. a _matching_ is stable when there is no match **(A,B) and (C,D)**  where **A** prefers **D** and **D** prefers **A** or any other combination where both **A** and **B** would be individually better off than they are with the element they are currently matched with.
+
+### Input Format
+- The first line is a set of key-value pairs of n men (k, v) where k is an integer greater than zero and v is a list of integers greater than zero of size n.
+- The second line is a set of key-value pairs of n women (k, v) where k is an integer greater than zero and v is a list of integers greater than zero of size n.
+
+### Constraints
+- Set of men and set of women must be equal in size.
+- Each match must be stable.
+- Each man must be engaged to a single woman and each woman to a single man.
+
+### Output Format
+- n rows of the form (M, W) where M is a male in input 1 who is engaged to woman W in input 2.
+
+## Algorithm
+### Overview
+We will use the Gale-Shapely algorithm, discovered in 1962 by David Gale and Lloyd Shapely. This algorithm proves that for any equal number of men and women it is always possible to solve the stable marriage problem.  
+
+Find a one-to-one correspondance between the set of men _m_ and women _w_ using the preferences of each man in _m_ and woman in _w_ such that each pairing is considered _stable_.
+
+Each element in each set will have a list of preferences mapping to each element in the other set, i.e. male M<sub>1</sub> in set _m_ will have a list [W<sub>1</sub>, W<sub>5</sub>m...W<sub>N</sub>] of preferred women in _w_ in decreasing order, as well as a status (single, engaged). The initial sets would look something like:
+
+<table>
+<tr><th> Set of Men </th><th> Set of Women </th></tr>
+<tr><td>
+
+| m  | Preferences  |  
+|----|--------------|
+| m<sub>1</sub> | [w<sub>2</sub>, w<sub>3</sub>,w<sub>4</sub>,w<sub>1</sub>] |
```

Comment:
Are you trying to do html tags inside source code markdown? Cause that won't work